src/rust/engine/process_execution/src/remote.rs

```diff
@@ -354,7 +354,16 @@ impl CommandRunner {
         })
         .to_boxed()
     } else {
-      future::ok(Bytes::from(execute_response.get_result().get_stdout_raw())).to_boxed()
+      let stdout_raw = Bytes::from(execute_response.get_result().get_stdout_raw());
+      let stdout_copy = stdout_raw.clone();
+      self
+        .store
+        .store_file_bytes(stdout_raw, true)
+        .map_err(move |error| {
+          ExecutionError::Fatal(format!("Error storing raw stdout: {:?}", error))
+        })
+        .wait();
```

Comment:
As discussed in slack, if you replace these waits with:
```
.map(|_| stdout_copy)
.to_boxed()
```
...then you should be good to go!

Adding a test would be good of course though.

Reply:
ah, nice. the `map(|_| stdout_copy)` was the bit I was missing. thanks!